### PR TITLE
add emojis to stored-list-filter and create ImageParam class

### PR DESCRIPTION
**Pull request by eddiemuc**
* add emojis to stored-list-filter and create ImageParam class
* also introduces the class "ImageParam" to pass image parameters to methods. 
* Follow-up idea óf second point would be to create similar classes/builders "TextParam", "ActionParam" etc and use these to reduce the necessary overloads in utility classes like "Dialogs" or "ViewUtils" which are currently necessary e.g. to pass Texts as id OR String

image

@moving-bits I would appreciate a review by you because I extended "`EmojiUtils`" and could really use your opinion on this.

### CODE REVIEW

**Comment by eddiemuc** [comment]
@moving-bits can I merge this one? I want to build up on it.

File: main/src/cgeo/geocaching/utils/EmojiUtils.java [hunk 1]
```diff
@@ -416,6 +422,30 @@ public static BitmapDrawable getEmojiDrawable(final EmojiPaint paint, final int
...
+    @NonNull
+    public static BitmapDrawable getEmojiDrawable(final int wantedSize, final int emoji) {
+
+        EmojiPaint p;
+        synchronized (EMOJI_PAINT_CACHE_PER_SIZE) {
+            p = EMOJI_PAINT_CACHE_PER_SIZE.get(wantedSize);
+            if (p == null) {
+                final Resources res = CgeoApplication.getInstance().getApplicationContext().getResources();
+                final Pair<Integer, Integer> temp = new Pair<>(wantedSize, wantedSize);
+                final Pair<Integer, Integer> markerDimensions = new Pair<>((int) (temp.first * 1.2), (int) (temp.second * 1.2));
```
Comment (moving-bits) [inline main/src/cgeo/geocaching/utils/EmojiUtils.java:440]: you don't need the intermediate step of creating `temp`, you can use `wantedSize * 1.2` directly in the second line

**Comment by moving-bits** [comment]
> @moving-bits can I merge this one? I want to build up on it.

I gave it a (short) glance yesterday, but did not fully understand all aspects, and had not found the time to test locally. But anyway, we are still in an early dev phase (no urgent need to release from `master`), and the change is local, so I would vote for merging, and I'll try to give it a more thorough look throughout the next days.

**Comment by eddiemuc** [comment]
Ok, thanks a lot for the (quick) review. I will correct your finding and merge afterwards

File: main/src/cgeo/geocaching/utils/EmojiUtils.java (hunk 1 above)
Comment (eddiemuc) [reply main/src/cgeo/geocaching/utils/EmojiUtils.java]: done

**Comment by eddiemuc** [comment]
retest this please

**Comment by eddiemuc** [comment]
failing test is unrelated, thus merging

**Pull request merged**